psl-evaluation/src/test/java/org/linqs/psl/utils/evaluation/statistics/DiscretePredictionComparatorTest.java

```diff
@@ -144,4 +144,44 @@ public void testAccuracy() {
 				assertEquals(26.0/30.0, acc, 1e-10);
 		}		
 	}
+
+	@Test
+	public void testPrecisionNegativeClass() {
+		for (double thresh = 0.1; thresh <= 1.0; thresh += 0.1) {
+			comparator.setThreshold(thresh);
+			DiscretePredictionStatistics comparison = comparator.compare(predicate, 6*5);
+			double prec = comparison.getPrecision(DiscretePredictionStatistics.BinaryClass.NEGATIVE);
+			if (thresh <= 0.8)
+				assertEquals(24.0/25.0, prec, 1e-10);
+			else
+				assertEquals(26.0/30.0, prec, 1e-10);
+		}
+	}
+	
+	@Test
+	public void testRecallNegativeClass() {
+		for (double thresh = 0.1; thresh <= 1.0; thresh += 0.1) {
+			comparator.setThreshold(thresh);
+			DiscretePredictionStatistics comparison = comparator.compare(predicate, 6*5);
+			double recall = comparison.getRecall(DiscretePredictionStatistics.BinaryClass.NEGATIVE);
+			if (thresh <= 0.8)
+				assertEquals(24.0/26.0, recall, 1e-10);
```

Comment:
1e-10 is a little too precise and may actually cause issues. I generally use 1e-5, unless there is a specific need for extra precision.